Khan/wonder-blocks
# Remove span from links without icons

## Summary:
Initially, the link text/`children` was nested inside a `span` element in order to use `verticalAlign` to center the text with icons.

Nesting `children` in a `span` broke other links that don't have icons (screenshots). 

This change checks first if icon props are present. If true, the text/`children` is nested inside a `span` element.

If the link has no icons, no `span` is needed.

Broken:

<img width="421" alt="Screenshot 2023-04-20 at 1 11 13 PM" src="https://user-images.githubusercontent.com/100858764/233461307-d3259fc9-9bb1-4e1f-a910-957f1e41a621.png">

From production:

<img width="420" alt="Screenshot 2023-04-20 at 1 11 06 PM" src="https://user-images.githubusercontent.com/100858764/233461287-a5254efe-cbcc-4f94-b3c4-6e58b9efa2d6.png">

Also removed `textUnderlineOffset` from `sharedStyles` because without the `span` element, the underline sat too low when inline:

![image](https://user-images.githubusercontent.com/100858764/233480412-119ef1d8-8f00-4c0f-89bc-8a9a5ba101a6.png)

Added `textUnderlineOffset` to `sharedStyles.withIcon` and `restingInline` to match the underline when inline:

<img width="576" alt="Screenshot 2023-04-20 at 4 35 58 PM" src="https://user-images.githubusercontent.com/100858764/233481931-5843a070-f58b-4e52-874d-f6be7a6d235b.png">


Issue: n/a

## Test plan:
For start and end icon props:
- In Storybook, inspect the Default `Link` story. 
- Confirm the link text is not nested inside a `span` element. 
- Add an icon via the control panel and notice in the DOM that a `span` element wraps the link text.

For external icon (since external icon relies on `target` prop:
- In Storybook, inspect the Default `Link` story. 
- Confirm the link text is not nested inside a `span` element. 
- Add "_blank" to `target` prop via the control panel and notice in the DOM that a `span` element wraps the link text.
